packages/components/upload/src/use-handlers.ts

```diff
@@ -166,11 +166,12 @@ export const useHandlers = (
     () => props.fileList,
     (fileList) => {
       for (const file of fileList) {
-        file.uid = genFileId()
+        file.uid ||= genFileId()
         file.status ||= 'success'
       }
+      uploadFiles.value = fileList as UploadFiles
```

Comment:
- Force type is not a good idea
- Why watch deep? The main usage case is restore previous uploaded files from api returns.